From 01a2010f1e832fe2f14144f176f4e849b82c3f25 Mon Sep 17 00:00:00 2001
From: Michael Jumper <mike.jumper@guac-dev.org>
Date: Wed, 11 Nov 2015 16:14:48 -0800
Subject: [PATCH] Recreate workers every 10 frames (see jnordberg/gif.js#20).

---
 src/gif.coffee | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/gif.coffee b/src/gif.coffee
index e87524d..6664ffa 100644
--- a/src/gif.coffee
+++ b/src/gif.coffee
@@ -100,6 +100,11 @@ class GIF extends EventEmitter
       worker = new Worker @options.workerScript
       worker.onmessage = (event) =>
         @activeWorkers.splice @activeWorkers.indexOf(worker), 1
+        if @finishedFrames % 10 == 0
+          freshWorker = new Worker @options.workerScript
+          freshWorker.onmessage = worker.onmessage
+          worker.terminate()
+          worker = freshWorker
         @freeWorkers.push worker
         @frameFinished event.data
       @freeWorkers.push worker
-- 
2.1.0

